### ISSUE TITLE
bug: Accounts pipeline often times out

### ISSUE THREAD

**Issue by mbevc1** [awslabs/landing-zone-accelerator-on-aws#556]
**Describe the bug**
When applying changes very often `Accounts` pipeline would break with timeouts changing resources.

```
Failed resources:
AWSAccelerator-AccountsStack-050451375726-us-east-1 | 10:42:41 AM | UPDATE_FAILED        | Custom::CreateServiceLinkedRole | SecurityHubServiceLinkedRole/CreateServiceLinkedRoleResource/Default (SecurityHubServiceLinkedRoleCreateServiceLinkedRoleResource4CC7EFAA) Received response status [FAILED] from custom resource. Message returned: TimeoutError: {"state":"TIMEOUT","reason":"Waiter has timed out"}
    at checkExceptions (/var/runtime/node_modules/@aws-sdk/node_modules/@smithy/util-waiter/dist-cjs/index.js:59:26)
    at waitUntilFunctionActiveV2 (/var/runtime/node_modules/@aws-sdk/client-lambda/dist-cjs/index.js:5826:49)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async defaultInvokeFunction (/var/task/outbound.js:1:1024)
    at async invokeUserFunction (/var/task/framework.js:1:2287)
    at async onEvent (/var/task/framework.js:1:369)
    at async Runtime.handler (/var/task/cfn-response.js:1:1676) (RequestId: ab51f95c-8b66-4b99-9420-f4b775751368)
    new CustomResource (/codebuild/output/src2823/src/s3/00/source/node_modules/aws-cdk-lib/core/lib/custom-resource.js:1:1075)
    \_ new ServiceLinkedRole (/codebuild/output/src2823/src/s3/00/source/packages/@aws-accelerator/constructs/lib/aws-iam/service-linked-role.ts:90:21)
    \_ AccountsStack.createServiceLinkedRole (/codebuild/output/src2823/src/s3/00/source/packages/@aws-accelerator/accelerator/lib/stacks/accelerator-stack.ts:918:31)
    \_ AccountsStack.createSecurityHubServiceLinkedRole (/codebuild/output/src2823/src/s3/00/source/packages/@aws-accelerator/accelerator/lib/stacks/accelerator-stack.ts:563:12)
    \_ new AccountsStack (/codebuild/output/src2823/src/s3/00/source/packages/@aws-accelerator/accelerator/lib/stacks/accounts-stack.ts:67:12)
    \_ createAccountsStack (/codebuild/output/src2823/src/s3/00/source/packages/@aws-accelerator/accelerator/utils/stack-utils.ts:502:27)
    \_ createManagementAccountStacks (/codebuild/output/src2823/src/s3/00/source/packages/@aws-accelerator/accelerator/bin/app.ts:105:22)
    \_ main (/codebuild/output/src2823/src/s3/00/source/packages/@aws-accelerator/accelerator/bin/app.ts:246:5)
    \_ processTicksAndRejections (node:internal/process/task_queues:95:5)
    \_ async /codebuild/output/src2823/src/s3/00/source/packages/@aws-accelerator/accelerator/bin/app.ts:260:5
```
**To Reproduce**
Apply changes to the manifest, I've noticed this usually with security related changes, e.g. GD, Macie,...

**Expected behavior**
Successful apply of the pipeline.

**Please complete the following information about the solution:**

- [x] Version: v1.9.2

> To get the version of the solution, you can look at the description of the created AWS CloudFormation stack used to install the LZA (AWSAccelerator-InstallerStack). For example, "__(SO0199) Landing Zone Accelerator on AWS. **Version 1.5.1.**__". If the description does not contain the version information, you can look at the Parameters of the stack for the **RepositoryBranchName** as that should contain the version number.

- [x] Region: [e.g. us-east-1]
- [x] Was the solution modified from the version published on this repository? No
- [ ] If the answer to the previous question was yes, are the changes available on GitHub?
- [ ] Have you checked your [service quotas](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html) for the services this solution uses?
- [ ] Were there any errors in the CloudWatch Logs?

**Screenshots**
If applicable, add screenshots to help explain your problem (please **DO NOT include sensitive information**).

**Additional context**
Possibly related to: https://github.com/awslabs/landing-zone-accelerator-on-aws/issues/237

**Comment by mbevc1** [comment]
There is also retry we could use as well: https://aws.amazon.com/about-aws/whats-new/2024/10/aws-codebuild-retrying-builds-automatically/

**Comment by n3mawashi** [comment]
Can confirm that I get his almost ever run. While I'm grateful for the retry action/stage, it usually means having force delete the template every run.

**Comment by hexmoose** [comment]
We had a very similar issue to this. The root cause was that our Lambda Concurrent Executions quota was too low (10). Setting this to the default value (1,000) fixed our frequent failures of the Accounts stack.

**Comment by richardkeit** [comment]
What version of the solution are you running?

**Comment by larsborasmussen** [comment]
Is there any information we can provide to give this issue more attention?

**Comment by Athanasii** [comment]
Thanks for opening an issue with LZA.  We are tracking this issue.  For immediate support and solutions, please reach out directly to [AWS support](https://docs.aws.amazon.com/solutions/latest/landing-zone-accelerator-on-aws/contact-aws-support.html).

**Comment by nagmesh** [comment]
Hi everyone, thanks for the continued reports and the community debugging.

Looking at the stack trace, the failure is in the CDK custom-resources Provider framework — specifically `waitUntilFunctionActiveV2`, which runs *before* the SLR Lambda is invoked. So while the resource being created is an IAM Service-Linked Role, the timeout is happening in the Lambda layer, not in `iam:CreateServiceLinkedRole`.

This matches the same root cause tracked in #984: AWS now ships new accounts with a default Lambda concurrent-executions quota of **10–50** (not the historical 1000). The Accounts stack creates multiple SLRs in parallel (SecurityHub, GuardDuty, Macie, Access Analyzer, etc.), each backed by a CDK Provider that fans out to several framework Lambdas. Under a 10-concurrency cap, those Lambdas contend for slots and the waiter eventually times out.

**To confirm this is what you're hitting, please check your Lambda concurrent-executions limit in the failing account/region:**

```bash
aws lambda get-account-settings --query AccountLimit.ConcurrentExecutions --region <region>
```

If the value is 10 or 50, request an increase to 1000 via AWS Support, or automate it for new accounts via the LZA `limits:` config:

```yaml
limits:
  - serviceCode: lambda
    quotaCode: L-B99A9384
    desiredValue: 1000
    deploymentTargets:
      organizationalUnits:
        - Root
```

(Note: the Service Quotas API requires `desiredValue` ≥ the global default of 1000 — see #984.)

**I'm closing this issue since it has been open for a while and the root cause is now well-understood.** If after raising your Lambda concurrency limit you are still seeing this failure, please open a new issue with:
- Output of the `get-account-settings` command above
- LZA version
- Whether retrying the failed pipeline action succeeds
